SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -2560,278 +2547,212 @@ void CDriver::Interface_Preprocessing(CConfig **config, CSolver***** solver, CGe
         continue;
       }
 
-      nMarkerInt = (int) ( config[donorZone]->GetMarker_n_ZoneInterface() / 2 );
+      const auto nMarkerInt = config[donorZone]->GetMarker_n_ZoneInterface() / 2;
 
       /*--- Loops on Interface markers to find if the 2 zones are sharing the boundary and to
        *--- determine donor and target marker tag ---*/
-      for (iMarkerInt = 1; iMarkerInt <= nMarkerInt; iMarkerInt++) {
-
-        markDonor  = -1;
-        markTarget = -1;
+      for (unsigned short iMarkerInt = 1; iMarkerInt <= nMarkerInt; iMarkerInt++) {
 
-        /*--- On the donor side ---*/
-        nMarkerDonor = config[donorZone]->GetnMarker_All();
-
-        for (iMarkerDonor = 0; iMarkerDonor < nMarkerDonor; iMarkerDonor++) {
-
-          /*--- If the tag GetMarker_All_ZoneInterface(iMarker) equals the index we are looping at ---*/
-          if ( config[donorZone]->GetMarker_All_ZoneInterface(iMarkerDonor) == iMarkerInt ) {
-            /*--- We have identified the identifier for the interface marker ---*/
-            markDonor = iMarkerDonor;
-
-            break;
-          }
-        }
-
-        /*--- On the target side ---*/
-        nMarkerTarget = config[targetZone]->GetnMarker_All();
-
-        for (iMarkerTarget = 0; iMarkerTarget < nMarkerTarget; iMarkerTarget++) {
-
-            /*--- If the tag GetMarker_All_ZoneInterface(iMarker) equals the index we are looping at ---*/
-          if ( config[targetZone]->GetMarker_All_ZoneInterface(iMarkerTarget) == iMarkerInt ) {
-              /*--- We have identified the identifier for the interface marker ---*/
-              markTarget = iMarkerTarget;
-
-              break;
-          }
+        /* --- Check if zones are actually sharing the interface boundary, if not skip. ---*/
```

Comment:
Maybe you can put these things that follow in a factory class ?

Reply:
I thought about it, but there's all kinds of complicated logic here with some classes requiring info from CSolver, it did not seem right to just copy paste the entire thing somewhere else.

Reply:
Done in #919